### ISSUE TITLE
[FEATURE REQUEST] Some requests about fmDHCP

### ISSUE THREAD

**Issue by seglavi** [WillyXJ/facileManager#632]
fM Version : **v4.6.2**
<Specify Module Name> Version : **v0.8.1**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/WillyXJ/facileManager/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/WillyXJ/facileManager/issues?q=is%3Aissue+is%3Aclosed) and [opened issues](https://github.com/WillyXJ/facileManager/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

---

Firstly, thank you very much for facileManager. I am using the fmDNS module, and it has truly made my life easier. I can't thank you enough.

I wanted to start using the fmDHCP module as well, but it seems that fmDHCP is somewhat overshadowed by the fmDNS module. Maybe it feels that way because I haven't spent much time with fmDHCP yet. Anyway, let me get to the point without taking up too much of your time:

In fmDHCP, the "authoritative" feature cannot be added as a Global Option. Shouldn't we be able to add this?

In the Leases section of fmDHCP, we can select and delete the displayed IPs, but wouldn't it be nice to have an option to completely clear all leases?

Lastly, let me mention this scenario:

I made some settings in the global options.
I added subnets, let's say 50 subnets.
Then I changed some values in the global options.
Despite my changes, the previous settings remain the same.
The subnets I add afterward are set according to the new values.
In this scenario, would it be too demanding to automate this process rather than requiring manual intervention?

Actually, I have many more things on my mind, but I want to give it some time and try to understand everything thoroughly. I apologize for taking up your time.

**Comment by WillyXJ** [comment]
> In fmDHCP, the "authoritative" feature cannot be added as a Global Option. Shouldn't we be able to add this?

Yes, this looks to be an oversight. You can define your subnets as authoritative, but the global option should also exist.

> In the Leases section of fmDHCP, we can select and delete the displayed IPs, but wouldn't it be nice to have an option to completely clear all leases?

Noted as a feature request.

> I made some settings in the global options. I added subnets, let's say 50 subnets. Then I changed some values in the global options. Despite my changes, the previous settings remain the same. The subnets I add afterward are set according to the new values. In this scenario, would it be too demanding to automate this process rather than requiring manual intervention?

This does seem like a bug that I'll dig into.

**Comment by WillyXJ** [comment]
> I made some settings in the global options.
> I added subnets, let's say 50 subnets.
> Then I changed some values in the global options.
> Despite my changes, the previous settings remain the same.
> The subnets I add afterward are set according to the new values.

Would you please provide more specific steps to reproduce this issue? So far, I'm unable to.

**Comment by rekodla** [comment]
Of course. First of all, thank you for your time. Please allow me to explain the scenario as follows:

1. Step: Config > Options > Global Options:
We entered our default values here.

2. Step: Objects > Networks > Subnets:
We entered our subnets here. While entering these subnets, the default values we set in step 1 appeared automatically, and we completed the entries accordingly.

3. Step: Changing Default Global Options Values:
Now, we changed the default global options values. For example, we modified the default-lease-time and max-lease-time values.

4. Step: Old Values Persisting in Subnets:
Until this point, the old global options values remain in the subnets we previously configured. If we want to apply the new global options values, we need to change them one by one in each subnet.

Looking at this entire scenario, is it a bug or a problem? No, it is not. Is it something that prevents work? Certainly not. However, if we have a large number of subnets, making changes to the global options values can be tedious. Therefore, I was just questioning if there could be a solution for this. I hope I have been clear enough. Thanks.

**Comment by seglavi** [comment]
Of course. First of all, thank you for your time. Please allow me to explain the scenario as follows:

Step: Config > Options > Global Options:
We entered our default values here.

Step: Objects > Networks > Subnets:
We entered our subnets here. While entering these subnets, the default values we set in step 1 appeared automatically, and we completed the entries accordingly.

Step: Changing Default Global Options Values:
Now, we changed the default global options values. For example, we modified the default-lease-time and max-lease-time values.

Step: Old Values Persisting in Subnets:
Until this point, the old global options values remain in the subnets we previously configured. If we want to apply the new global options values, we need to change them one by one in each subnet.

Looking at this entire scenario, is it a bug or a problem? No, it is not. Is it something that prevents work? Certainly not. However, if we have a large number of subnets, making changes to the global options values can be tedious. Therefore, I was just questioning if there could be a solution for this. I hope I have been clear enough. Thanks.

**Comment by WillyXJ** [comment]
Thanks for the steps as I'm able to reproduce it now. However, looking at the behavior and the intention, you shouldn't even get those fields automatically populated during new forms. :)  The intention is "Global Options" would be config settings that would apply to all subnets, hosts, etc. and then defining them in a subnet or other object would override the global option for that one object.  This is how dhcpd.conf is designed (based on the docs I've read and experiences).  It's not meant to be a template.

So, it is indeed a bug and the next release will remove the automatic population in the forms.  By deleting the values in your object forms, those objects will inherit the global option values when dhcpd is reloaded.

**Comment by seglavi** [comment]
First of all, thank you once again. As I mentioned in my initial message, fmDNS has been incredibly helpful and has significantly reduced my workload. This led me to think about whether we could do something similar for fmDHCP, and I shared some of my experiences with you. Reflecting on the last example we discussed, I must clarify that I'm not insisting this is a problem based on any specific expertise or experience. My only thought was this: as you mentioned, the global options are stored in the database when adding a subnet. When I change the global settings, the database doesn't update, so the old values remain there. This wouldn't be an issue if there were only a few subnets, but with 100 subnets, it does take time (I know because I've tried it :)).

In conclusion, thank you again and again for this wonderful work. It is so well done and so useful, I can't express it enough. You are amazing!

**Comment by WillyXJ** [comment]
This is now implemented in [fmDHCP 0.9.0](http://www.facilemanager.com/modules/fmdhcp/) and later.
